crates/oxc_minifier/src/plugins/replace_global_defines.rs

```diff
@@ -125,53 +146,50 @@ impl<'a> ReplaceGlobalDefines<'a> {
         }
     }
 
-    fn replace_dot_defines(&self, expr: &mut Expression<'a>) {
+    fn replace_dot_defines(&mut self, expr: &mut Expression<'a>) {
         if let Expression::StaticMemberExpression(member) = expr {
-            'outer: for (parts, value) in &self.config.0.dot_defines {
-                assert!(parts.len() > 1);
-
-                let mut current_part_member_expression = Some(&*member);
-                let mut cur_part_name = &member.property.name;
+            for dot_define in &self.config.0.dot_defines {
+                if Self::is_dot_define(dot_define, member) {
+                    let value = self.parse_value(&dot_define.value);
```

Comment:
Parsing each time you replace is no longer required. `CloneIn` now exists!

But the tricky part is (as usual) scopes. If the replacement can be an expression of arbitrary complexity, it may contain scopes, internal references, or external references. This is really hard.

If you're keen to get this in asap, but want it to be correct, probably need to run semantic again after this pass.

But... could check if the replacement expression contains scopes or references, and if it doesn't you can skip that extra semantic pass. Skipping is possible for the most common use cases e.g. replacing `DEBUG` with `true`.